internal/filter/filter.go

```diff
@@ -169,6 +211,67 @@ func (f *Filter) FindModulesForNode(ctx context.Context, node client.Object) []r
 	return reqs
 }
 
+func (f *Filter) FindModulesForNMCNodeChange(ctx context.Context, node client.Object) []reconcile.Request {
+	logger := ctrl.LoggerFrom(ctx).WithValues("node", node.GetName())
+
+	reqs := make([]reconcile.Request, 0)
+
+	logger.Info("Listing all modules")
+
+	mods := kmmv1beta1.ModuleList{}
+
+	err := f.client.List(context.Background(), &mods)
+	if err != nil {
+		logger.Error(err, "could not list modules")
+		return reqs
+	}
+
+	logger.Info("Listed modules", "count", len(mods.Items))
+
+	reqSet := sets.New[types.NamespacedName]()
+
+	for _, mod := range mods.Items {
+		logger := logger.WithValues("module name", mod.Name)
+
+		logger.V(1).Info("Processing module")
+
+		moduleSelectorMatchNode, err := utils.IsObjectSelectedByLabels(node.GetLabels(), mod.Spec.Selector)
+		if err != nil {
+			logger.Error(err, "could not determine if node is selected by module", "node", node.GetName(), "module", mod.Name)
+			return reqs
+		}
+
+		if !moduleSelectorMatchNode {
+			logger.V(1).Info("Node labels do not match the module's selector; skipping")
+			continue
+		}
+
+		nsn := types.NamespacedName{Name: mod.Name, Namespace: mod.Namespace}
+
+		reqs = append(reqs, reconcile.Request{NamespacedName: nsn})
+		reqSet.Insert(nsn)
+	}
+
+	nms, err := f.nmcHelper.Get(ctx, node.GetName())
+	if err != nil || nms == nil {
+		return reqs
+	}
+
+	// go over modules of NodeModulesConfig and add them to request if they are not there already
```

Comment:
Why should we do this? We have enqueued all Modules that we need to reconcile already.
Besides, this function should return NamespacedNames for Modules only, not Nodes or NMCs.

Reply:
1) because we should also find the modules that should be reconciled in the opposite direction ( they should not run on the node anymore. If Node's labels were changed/removed, the first part of the function will find the modules that should run on the node, but we must also check the already running Modules (stored in the NMC) , in case they need to be removed
2) the functions does return NamespacedName of the modules, which are stored in nmc.Spec.Modules